core/storage/btree.rs

```diff
@@ -1122,6 +1122,36 @@ impl BTreeCursor {
             .copy_from_slice(&buffer[..num_bytes as usize]);
     }
 
+    /// Check if any ancestor pages still have cells to iterate.
+    /// If not, traversing back up to parent is of no use because we are at the end of the tree.
+    fn ancestor_pages_have_more_children(&self) -> Result<CursorResult<bool>> {
```

Comment:
should we store parent cell count together with `cell_indices` in `PageStack` so that this shit is not required...?